### DISCUSSION TITLE
Concerning Privacy: add options to choose what could be shown on the contribution activity on the profile page

### DISCUSSION THREAD

**Discussion by grashaar** [community/community#4098] [Other Feature Feedback, Questions, & Ideas]
[...] and how much details we want to show; or disable it entirely from the eye of other viewers.
And any API that requests this kind of information should respect the options we have set.

---

> 
> 
> i don't like it. i don't want people to be able to crawl that info, either, even through an API. it gets spammy when you make a lot of typo commits, PR merges, or open a lot of issues.
> 
> i don't mind the activity tab, but being able to see what people have done over the past year on any given date is creepy.
> 
> what i would really like is the two columns of `popular repositories` and `repositories contributed to` to be really long. way more interesting IMO. the `repositories` tab requires too much scrolling.

Mirgrate from https://github.com/isaacs/github/issues/142

It seems the issue persisted over 7 years now should be addressed seriously. It is the right time to fix this, when privacy is the main concern of the internet world today.

Whether it is for the community or a renowned organization, I don't want people to track me around.

---

Edit 1:
A related issue from 2016 https://github.com/isaacs/github/issues/627

It surprises me that people has been raising this issue over all those years but nothing changes.

**Comment by erdo** [comment]
Most people are probably aware that this data could be compiled by a 3rd party and re-published, that's not the issue (at least for me). The ease with which this data is available to a casual observer of my page (my clients or employers for instance) is the thing that I dislike, and not having the autonomy to hide that data, even though I pay for my account is bizarre and insulting.

**Comment by ghost** [comment]
I also have been tracked. Someone saw my issue regarding reverse enginnering, and the original dev came to me and asked why I need his codes lol. I was surprised lol, so I had create a dummy account.
I hope this feature will be added ASAP

**Reply by whizzzkid** [reply]
I actually faced something similar recently, had to make my account private. Scary!

**Comment by shenlebantongying** [comment]
Other Reasons: 

- Don't want to others know their "spam" contributions like fixing typos, translations, and mini fixes isaacs/github#142
- Don't want to annoy their employer isaacs/github#627
- Want more privacy isaacs/github#1840
- Dislike gamification isaacs/github#1234
- Incorrect representation for open source contributions isaacs/github#1290
- etc.

**Reply by Levi-Lesches** [reply]
These issues had tons of support over at the old repo -- 1,750 upvotes between the issues linked above. GitHub, please take that into consideration here.

**Comment by LuciusV** [comment]
This is ridiculous, how you dare to not listen to community to that degree?

**Comment by crystalcommunication** [comment]
I agree with pretty much everything above. I understand that most of my contributions on GitHub are public anyway, and could easily be aggregated by a third party service crawling the site, but I find it deeply disturbing that anyone who stumbles across my GitHub profile can gather a great deal of information about me simply by clicking on my username and looking at the commits I've made to third-party repositories and issues I've opened. I don't care about internet clout, I just use this account to contribute to other projects that have chosen to use GitHub and I should have control over what information people can see on my profile page.

**Reply by thewhitegrizzli** [reply]
exactly my thoughts

**Comment by AndresSweeneyRios** [comment]
This is an important privacy issue IMO, why hasn't this been addressed yet?

**Comment by dingpanbao** [comment]
n years passed

**Comment by rt-hamilton** [comment]
Besides the already mentioned privacy concerns, one should also consider harmful consequences of (unnecessary?) gamification. GitHub already removed an existing profile "feature" some years ago, namely streaks. I don't think that they ever explained that decision publicly, but I guess that GitHub is aware of this issue. There are insights on how that design feature and the design change to user profiles influenced the behavior of developers on GitHub (see https://arxiv.org/pdf/2006.02371.pdf and https://github.com/lukasmoldon/GHStreaksThesis):

_"The observed effects of the removal of the counters implies that platform designers have some responsibility to consider how the introduction of gamification elements steers behavior. [...] Our findings suggest that some users will change their behavior to collect digital tokens, but that this behavior may optimize for the game, and not necessarily for healthy and productive activity."_

I think that this also applies to the remaining contribution calendar.

**Comment by ghost** [comment]
cirosantilli mentioned on the issue https://github.com/isaacs/github/issues/1840 that GitHub admins actually already have a button to hide contribution activity. [His profile](https://github.com/cirosantilli) ([archived](https://web.archive.org/web/20220131152623/https://github.com/cirosantilli)) still have entire contribution activity hidden as of 2022

Since Github still hasn't read this discussions, should we all contact Github to add an option to hide contribution activity for everyone?

**Comment by evi-liu** [comment]
👋 Hi folks, thanks for all of the feedback on this. We recently shipped Private Profiles, which enables you to hide your contributions, activity, and social features from your GitHub profile. 

You can read more at our Changelog: https://github.blog/changelog/2022-04-21-private-profiles/

**Reply by grashaar** [reply]
Hi. I welcome the attempt to fix this. But I won't call this a good answer to the issue. Certainly I don't want my profile to be private, I want people to see my pins on the overview page. But I also want to hide the contribution activity and graph from the public eyes. Profile overview is fine, but the contribution activity and graph is not.

**Reply by ghost** [reply]
> Hi. I welcome the attempt to fix this. But I won't call this a good answer to the issue. Certainly I don't want my profile to be private, I want people to see my pins on the overview page. But I also want to hide the contribution activity and graph from the public eyes. Profile overview is fine, but the contribution activity and graph is not.

I agree with you, I only want to hide my contribution activity

**Reply by ronaldaug** [reply]
It'd be great without hiding pinned repos. The rest are just fine. 👍

**Reply by z4cki** [reply]
yep, I only want to hide my contribution activity +1

**Reply by ghost** [reply]
I like it. There is always room for improvement, but I appreciate that we now have this option. Thanks @evi-liu!

**Comment by wassimchouchen** [comment]
take a llook this may help you understanding the hole issue : https://github.blog/changelog/2022-04-21-private-profiles/

**Comment by whizzzkid** [comment]
The [original issue](https://github.com/isaacs/github/issues/142) was posted 9+ years ago and I am surprised it's took the maintainers this long to arrive to a conclusion.

The profile pages are essentially broken, the [private profile](https://github.com/orgs/community/discussions/4098#discussioncomment-2656752) feature is all or nothing. I would like to show my pins, my achievements, devs I sponsor, even the contribution graph. The contribution activity on the other hand is a bit too verbose for me, creepy, if I must say. The only solution is to make my profile private. That makes everything private (even if other options like show achievements is on), that's a bad UX.

I would like to understand what's missing to implement a toggle on per component basis, we're all devs maybe we can help?

Please fix this or ask community for help!

**Reply by whizzzkid** [reply]
At least we're on Page 1 for issues that have not been answered and are not feedback for beta: https://github.com/orgs/community/discussions?discussions_q=sort%3Atop+is%3Aunanswered+-label%3ABeta+ I am not even sure if @github-staff is active on these issues or uses these to prioritize their work.

**Comment by strlns** [comment]
I'm especially infuriated that GitHub forcibly collects and publishes a push-history, independent from the repository graph.

This makes sense for professional PR workflows, but should be opt-out and controllable by the repository owner.

Git repositories by themselves allows to "rewrite history".

But the "activity" widget shows multiple "commits" when amending and force-pushing repeatedly.

No where have I been told that GitHub publishes information about me that is not even in git index of the corresponding reposity.
This feels like it could almost be illegal in Europe. [^1]

What's next, a "currently online" or "last online" display without possibility to opt out?

GitHub needs more competition, soon. 
This makes me start to regret choosing GitHub free tier at all for personal projects.
Unfortunately the network effect makes it very compelling.

[1] Addendum:

It might also be a sign that using GitHub for personal projects is simply a bad idea.

It should be clear to everyone at this point that the code you push to GitHub is not fully owned by you.

Your code snippets or derived work might be regurgitated at any time by Copilot or other MS products, derived DNN weights might even be sold.

For public repositories, the problem I described above is similar to edit histories for discussion posts. Opinions may differ a lot on what is sensible.

**Comment by whizzzkid** [comment]
One more thing, this info is parsed by google too:

![Screenshot 2023-09-12 at 2 40 55 PM](https://github.com/community/community/assets/1895906/5cfe7806-f732-4b05-a415-3afde130f79a)

So unless your info is private, this shows up on google search, what an invasion of privacy.

**Comment by MarcvdSluys** [comment]
This issue seems to have been around for about 10 years (at least), and I won't be surprised if it will be for another decade.  Meanwile, I have decided to write a small Python script to confuse the Contributions calendar with fake git commits:  https://github.com/MarcvdSluys/fake-commits

**Reply by whizzzkid** [reply]
make this a github action that auto-triggers.

**Comment by subham8907** [comment]
can we make tool get accurate contributors
